pkg/crypto/keys/private_key.go

```diff
@@ -20,16 +20,21 @@ type PrivateKey struct {
 	ecdsa.PrivateKey
 }
 
-// NewPrivateKey creates a new random Secp256k1 private key.
+// NewPrivateKey creates a new random Secp256r1 private key.
 func NewPrivateKey() (*PrivateKey, error) {
-	priv, x, y, err := elliptic.GenerateKey(elliptic.P256(), rand.Reader)
+	return NewPrivateKeyOnCurve(elliptic.P256())
+}
+
+// NewPrivateKeyOnCurve creates a new random private key using curve c.
+func NewPrivateKeyOnCurve(c elliptic.Curve) (*PrivateKey, error) {
```

Comment:
I'm not sure we can safely allow any `elliptic.Curve` here. We really only work with r1 and k1, so maybe it's enough to make this internal and expose `NewK1PrivateKey()` along with the old `NewPrivateKey` to the outside world.

Reply:
Fixed.